File: app/src/test/kotlin/tech/mmarca/openvitals/data/local/beverage/BeverageEntityTest.kt
package tech.mmarca.openvitals.data.local.beverage

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import tech.mmarca.openvitals.domain.model.CaffeineSourceCategory

class BeverageEntityTest {
    @Test
    fun `preloaded defaults include water category drinks`() {
        val waters = BeverageEntity.preloadedDefaults()
            .take(2)
            .map(BeverageEntity::toDomain)

        assertEquals(listOf("openvitals-still-water", "openvitals-gasified-water"), waters.map { it.id })
        assertEquals(listOf("Still water", "Gasified water"), waters.map { it.name })
        waters.forEach { water ->
            assertEquals(100.0, water.volumeMilliliters, 0.001)
            assertEquals(1.0, water.hydrationMultiplier, 0.001)
            assertEquals(CaffeineSourceCategory.WATER, water.category)
            assertEquals(true, water.isPreloaded)
            assertTrue(water.nutrientValues.isEmpty())
        }
    }

    @Test
    fun `preloaded defaults seed the caffeine catalog after the waters`() {
        val defaults = BeverageEntity.preloadedDefaults()

        // Sort orders are contiguous starting at 0 (waters), then the catalog.
        assertEquals(0, defaults[0].sortOrder)
        assertEquals(1, defaults[1].sortOrder)
        assertEquals(2, defaults[2].sortOrder)
        assertTrue(defaults.size > 2)
        // Catalog presets carry the caffeinehealth- id prefix and are preloaded.
        val firstPreset = defaults[2]
        assertTrue(firstPreset.id.startsWith("caffeinehealth-"))
        assertTrue(firstPreset.isPreloaded)
        assertFalse(firstPreset.isDeleted)
    }
}

File: test/data/local/beverage/beverage_entity_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:openvitals/data/local/beverage/beverage_entity.dart';
import 'package:openvitals/domain/model/caffeine_models.dart';

void main() {
  group('BeverageEntity', () {
    test('preloaded defaults include water category drinks', () {
      final waters = BeverageEntity.preloadedDefaults()
          .take(2)
          .map((entity) => entity.toDomain())
          .toList();

      expect(
        waters.map((water) => water.id).toList(),
        ['openvitals-still-water', 'openvitals-gasified-water'],
      );
      expect(
        waters.map((water) => water.name).toList(),
        ['Still water', 'Gasified water'],
      );
      for (final water in waters) {
        expect(water.volumeMilliliters, closeTo(100.0, 0.001));
        expect(water.hydrationMultiplier, closeTo(1.0, 0.001));
        expect(water.category, CaffeineSourceCategory.water);
        expect(water.isPreloaded, isTrue);
        expect(water.nutrientValues, isEmpty);
      }
    });

    test('preloaded defaults seed the caffeine catalog after the waters', () {
      final defaults = BeverageEntity.preloadedDefaults();

      // Sort orders are contiguous starting at 0 (waters), then the catalog.
      expect(defaults[0].sortOrder, 0);
      expect(defaults[1].sortOrder, 1);
      expect(defaults[2].sortOrder, 2);
      expect(defaults.length, greaterThan(2));
      // Catalog presets carry the caffeinehealth- id prefix and are preloaded.
      final firstPreset = defaults[2];
      expect(firstPreset.id, startsWith('caffeinehealth-'));
      expect(firstPreset.isPreloaded, isTrue);
      expect(firstPreset.isDeleted, isFalse);
    });
  });
}
